native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/NativeImagePlugin.java

```diff
@@ -271,6 +273,21 @@ private void configureJavaProject(Project project, Provider<NativeImageService>
             task.getMergeWithExisting().set(graalExtension.getAgent().getMetadataCopy().getMergeWithExisting());
             task.getToolchainDetection().set(graalExtension.getToolchainDetection());
         });
+
+        GraalVMReachabilityMetadataRepositoryExtension metadataRepositoryExtension = reachabilityExtensionOn(graalExtension);
+        TaskCollection<ReachabilityMetadataCopyTask> reachabilityMetadataCopyTasks = project.getTasks()
```

Comment:
:x: I think I would change this to basically _always_ register the task. Currently the configuration works a bit backwards: it waits for the user to create a task of type `ReachabilityMetadataCopyTask` and then configures it. This isn't quite Gradle idiomatic: I think the plugin should register the task, always, and configure some defaults (convention). This would also simplify documentation.

Reply:
I'll add a "nativeReachabilityMetadata" (most of the other task names start with "native") then let the conventions apply as before.